src/EditorFeatures/Core.Wpf/Options/EditorConfigDocumentOptionsProvider.Ide.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.CodingConventions;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.Options
+{
+    internal sealed partial class EditorConfigDocumentOptionsProvider
+    {
+        private static readonly ConditionalWeakTable<CodingConventionsChangedEventArgs, HashSet<ProjectId>> s_projectNotifications =
+            new ConditionalWeakTable<CodingConventionsChangedEventArgs, HashSet<ProjectId>>();
+
+        partial void OnCodingConventionContextCreated(DocumentId documentId, ICodingConventionContext context)
+        {
+            context.CodingConventionsChangedAsync += (sender, e) => HandleCodingConventionsChangedAsync(documentId, sender, e);
+        }
+
+        private Task HandleCodingConventionsChangedAsync(DocumentId documentId, object sender, CodingConventionsChangedEventArgs e)
+        {
+            var projectId = documentId.ProjectId;
+            var projectsAlreadyNotified = s_projectNotifications.GetOrCreateValue(e);
+            lock (projectsAlreadyNotified)
+            {
+                if (!projectsAlreadyNotified.Add(projectId))
+                {
+                    return Task.CompletedTask;
+                }
+            }
+
+            var foregroundNotificationService = ((IMefHostExportProvider)_workspace.Services.HostServices).GetExports<IForegroundNotificationService>().Single().Value;
+            foregroundNotificationService.RegisterNotification(UpdateProject, _listener.BeginAsyncOperation(nameof(HandleCodingConventionsChangedAsync)), CancellationToken.None);
+
+            return Task.CompletedTask;
+
+            void UpdateProject()
+            {
+                var compilationOptions = _workspace.CurrentSolution.GetProject(projectId)?.CompilationOptions;
+                if (compilationOptions == null)
+                {
+                    return;
+                }
+
+                var parseOptions = _workspace.CurrentSolution.GetProject(projectId).ParseOptions;
+                if (parseOptions.Features.TryGetValue("EditorConfigWorkaround", out _))
+                {
+                    parseOptions = parseOptions.WithFeatures(parseOptions.Features.Where(feature => feature.Key != "EditorConfigWorkaround"));
+                }
+                else
+                {
+                    parseOptions = parseOptions.WithFeatures(parseOptions.Features.Concat(new[] { KeyValuePair.Create("EditorConfigWorkaround", "true") }));
+                }
+
+                _workspace.OnCompilationOptionsChanged(projectId, compilationOptions);
```

Comment:
why do we need compilation options?

Reply:
It's possible we do not. This is modeled after some project system code following a similar pattern:

https://github.com/dotnet/roslyn/blob/d0ffbbd6ddfec1cd588b9c4e0f08e369267910c2/src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject_Options.cs#L131-L132

Reply:
I think compilation option won't matter here. and you can remove it.

Reply:
ah. this probably need +1 from @jasonmalinowski 

I dont believe we update Workspace.OnCompilation or OnParseOption any other place except ProjectSystem.

this now basically make workspace state to be inconsistent with one in project system.  

this at least maintain consistency between Project, ProjectState, ProjectInfo by going thorough workspace.

Reply:
@sharwell is correct that models what other code does; it's also correct that we don't consider that code good. TBH I am deeply saddened by the protected internalness of those methods, beause it means we get to cheat on our own problematic API. (I'm also still abusing this in my new PR!)

In any case, this code is still worrisome to me. There's no clean synchronization: if between the fetching of the options and the call to On* we did another change to the workspace, this is blowing away that change. This code is marginally safe in that it's hopping to the UI thread so it's safe up until I merge my free-threaded PR and then this is broken again.

Going to play my token card here: why are we doing it this way? What are we really trying to do?

Reply:
(given this will also force us to re-parse all source files, oof this seems expensive.)

Reply:
> (given this will also force us to re-parse all source files, oof this seems expensive.)

We probably don't need to reparse, but we do need to re-analyze the whole project. I couldn't find a way to separate the two on our timeline.

Reply:
@heejaechang's previous attempt at supporting .editorconfig refresh added a new "options changed" primitive to the workspace that was usable for this. We should resurrect that.